api/src/main/java/org/pmiops/workbench/cdr/model/Criteria.java

```diff
@@ -0,0 +1,235 @@
+    public long getSortOrder() {
+        return sortOrder;
+    }
+
+    public void setSortOrder(long sortOrder) {
+        this.sortOrder = sortOrder;
+    }
+
+    public Criteria sortOrder(long sortOrder) {
+        this.sortOrder = sortOrder;
+        return this;
+    }
+
+    @Column(name = "parent_id")
+    public long getParentId() {
+        return parentId;
+    }
+
+    public void setParentId(long parentId) {
+        this.parentId = parentId;
+    }
+
+    public Criteria parentId(long parentId) {
+        this.parentId = parentId;
+        return this;
+    }
+
+    @Column(name = "type")
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public Criteria type(String type) {
+        this.type = type;
+        return this;
+    }
+
+    @Column(name = "code")
+    public String getCode() {
+        return code;
+    }
+
+    public void setCode(String code) {
+        this.code = code;
+    }
+
+    public Criteria code(String code) {
+        this.code = code;
+        return this;
+    }
+
+    @Column(name = "name")
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Criteria name(String name) {
+        this.name = name;
+        return this;
+    }
+
+    @Column(name = "is_group")
+    public boolean getGroup() {
+        return group;
+    }
+
+    public void setGroup(boolean group) {
+        this.group = group;
+    }
+
+    public Criteria group(boolean group) {
+        this.group = group;
+        return this;
+    }
+
+    @Column(name = "is_selectable")
+    public boolean getSelectable() {
+        return selectable;
+    }
+
+    public void setSelectable(boolean selectable) {
+        this.selectable = selectable;
+    }
+
+    public Criteria selectable(boolean selectable) {
+        this.selectable = selectable;
+        return this;
+    }
+
+    @Column(name = "est_count")
+    public String getCount() {
+        return count;
+    }
+
+    public void setCount(String count) {
+        this.count = count;
+    }
+
+    public Criteria count(String count) {
+        this.count = count;
+        return this;
+    }
+
+    @Column(name = "concept_id")
+    public String getConceptId() {
+        return conceptId;
+    }
+
+    public void setConceptId(String conceptId) {
+        this.conceptId = conceptId;
+    }
+
+    public Criteria conceptId(String conceptId) {
+        this.conceptId = conceptId;
+        return this;
+    }
+
+    @Column(name = "domain_id")
+    public String getDomainId() {
+        return domainId;
+    }
+
+    public void setDomainId(String domainId) {
+        this.domainId = domainId;
+    }
+
+    public Criteria domainId(String domainId) {
+        this.domainId = domainId;
+        return this;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```

Comment:
You could look at using EqualsBuilder and HashCodeBuilder here:

https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/builder/EqualsBuilder.html

Or you can at least use Objects.equals() for the nullable fields.

Reply:
sure I can change them.. i let intellij generate these automatically for me and i never really pay attention to them.

Reply:
OK, I didn't know it supported that. 

I don't want to create extra work here... either way is fine by me.